From 4cb72b2c9e1dc87319f8aeb0b20e70039f25fddb Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Fri, 29 Aug 2014 15:41:55 +0800
Subject: [PATCH 0086/1594] ENGR00329334 ARM: IMX6SX: PXP: add dtb support for
 PXP in 3.14 kernel branch

commit f10a46507b122eb32bf77f781c95f7e3f1db86f4 from
git://git.freescale.com/imx/linux-2.6-imx.git

Add the dtb support for PXP to the new kernel branch
3.14.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 arch/arm/boot/dts/imx6sx-sdb.dtsi |    4 ++++
 arch/arm/boot/dts/imx6sx.dtsi     |    1 +
 2 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx-sdb.dtsi b/arch/arm/boot/dts/imx6sx-sdb.dtsi
index bf02892..e17f75c 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dtsi
+++ b/arch/arm/boot/dts/imx6sx-sdb.dtsi
@@ -311,6 +311,10 @@
 	status = "okay";
 };
 
+&pxp {
+	status = "okay";
+};
+
 &uart1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart1>;
diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index 1391f80..76126c7 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -1107,6 +1107,7 @@
 				};
 
 				pxp: pxp@02218000 {
+					compatible = "fsl,imx6sx-pxp-dma", "fsl,imx6sl-pxp-dma", "fsl,imx6dl-pxp-dma";
 					reg = <0x02218000 0x4000>;
 					interrupts = <GIC_SPI 8 IRQ_TYPE_LEVEL_HIGH>;
 					clocks = <&clks IMX6SX_CLK_PXP_AXI>,
-- 
1.7.5.4

